iter_gen_homework/iterators.py

```diff
@@ -0,0 +1,82 @@
+class SimpleIterator:
+    def __init__(self, iterable):
+        self.iterable = iterable
+        self.index = 0
+
+    def __next__(self):
+        try:
+            val = self.iterable[self.index]
+            self.index += 1
+            return val
+        except Exception:
+            raise StopIteration
+
+    def __iter__(self):
+        return self
+
+
+class ReversedIterator:
+
+    def __init__(self, iterable):
+        self.iterable = iterable
+
+    def __next__(self):
+        ...
+
+    def __iter__(self):
+        yield from self.iterable[::-1]
+
+
+class CycleIterator:
+    def __init__(self, iterable):
+        self.iterable = iterable
+        self.index = 0
+
+    def __next__(self):
+        val = self.iterable[self.index]
+        self.index = 0 if self.index == len(self.iterable) - 1 else self.index + 1
+        return val
```

Comment:
FYI 
as you can see Item to be iterable usually should implement   __len__ and __getitem__  methods

Reply:
How can I assure that in code? Just use smth like: `def __init__(self, iterable: typing.Iterable):` ?